### PR TITLE
Merge TypeDB Python driver into TypeDB driver repository

### PR DESCRIPTION
**Pull request by dmitrii-ubskii**
## What is the goal of this PR?

We merge driver-python into the main typedb-driver repository.

## What are the changes implemented in this PR?

- move the Python driver implementation into the top-level python/ package;
-  merge the two git trees.

### CODE REVIEW

File: python/BUILD [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+)
+
+deploy_pip(
+    name = "deploy-pip",
+    target = ":assemble-pip",
+    snapshot = deployment["pypi.snapshot"],
+    release = deployment["pypi.release"],
+)
+
+deploy_github(
```
[the remark this answers is not in the document]
Comment (dmitrii-ubskii) [reply python/BUILD:113]: These build just fine.

File: python/BUILD [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    draft = False
+)
+
+release_validate_python_deps(
+    name = "release-validate-python-deps",
+    requirements = "//python:requirements.txt",
+    tagged_deps = [
+        "typedb-protocol",
+    ],
+)
```
[the remark this answers is not in the document]
Comment (dmitrii-ubskii) [reply python/BUILD:129]: We still have a dependency on "parse", so probably?

File: python/README.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# TypeDB Client for Python
```
[the remark this answers is not in the document]
Comment (dmitrii-ubskii) [reply python/README.md:1]: Kinda?

File: python/BUILD (hunk 2 above)
[the remark this answers is not in the document]
Comment (flyingsilverfin) [reply python/BUILD:129]: yes ok, just without `typedb-protocol` right?

**Pull request merged**